serverless/src/layer.ts

```diff
@@ -140,21 +144,30 @@ export function applyLayers(
       }
 
       log.debug(`Setting Node Lambda layer for ${lambda.key}`);
-      layerARN = getLayerARN(region, nodeLayerVersion, lambda.runtime);
+      lambdaLibraryLayerArn = getLayerARN(region, nodeLayerVersion, lambda.runtime);
+      addLayer(lambdaLibraryLayerArn, lambda);
     }
 
-    if (layerARN !== undefined) {
-      const currentLayers = lambda.properties.Layers ?? [];
-      if (!new Set(currentLayers).has(layerARN)) {
-        currentLayers.push(layerARN);
-      }
-      lambda.properties.Layers = currentLayers;
+    if (extensionLayerVersion !== undefined) {
+      log.debug(`Setting Lambda Extension layer for ${lambda.key}`);
+      lambdaExtensionLayerArn = getLayerARN(region, extensionLayerVersion, "LambdaExtension");
+      addLayer(lambdaExtensionLayerArn, lambda);
     }
   });
   return errors;
 }
 
-function getLayerARN(region: string, version: number, runtime: string) {
+function addLayer(layerArn: string, lambda: LambdaFunction) {
+  if (layerArn !== undefined) {
+    const currentLayers = lambda.properties.Layers ?? [];
+    if (!new Set(currentLayers).has(layerArn)) {
```

Comment:
nit: a bit more compact/efficient to do this
```suggestion
    if (!currentLayers.includes(layerArn)) {
```

Reply:
modified it in latest commit